src/Twig/Node/DeprecatedTemplateNode.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Twig\Node;
+
+use Twig\Compiler;
+use Twig\Node\Expression\AbstractExpression;
+use Twig\Node\Node;
+
+class DeprecatedTemplateNode extends Node
+{
+    public function __construct(AbstractExpression $oldTemplate, AbstractExpression $newTemplate, $line, $tag = null)
+    {
+        parent::__construct(['oldTemplate' => $oldTemplate, 'newTemplate' => $newTemplate], [], $line, $tag);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function compile(Compiler $compiler)
+    {
+        @trigger_error(sprintf(
+            'The "%s" template is deprecated. Use "%s" instead.',
+            $this->getNode('oldTemplate')->getAttribute('value'),
```

Comment:
Do we need the `oldTemplate`? Can we resolve this automatically with `$this->getTemplateName()` like in
https://github.com/sonata-project/SonataCoreBundle/blob/11e833fbcf04a108b6b1606344ffa2819505a15f/src/Twig/Node/TemplateBoxNode.php#L82?

Reply:
@core23 removed, will add tests tomorrow, where should I add documentation for this?